core/src/main/scala/chisel3/VerificationStatement.scala

```diff
@@ -43,12 +43,12 @@ object assert extends VerifPrintMacrosDoc {
   // Macros currently can't take default arguments, so we need two functions to emulate defaults.
   def apply(
     cond:    Bool,
-    message: String,
+    message: => String,
```

Comment:
In the Pull Request description you said:

> This is to be backported to 3.5.x as an error instead of a warning

I think you got that backwards that it should be backported as a warning, not an error.

We have a problem though, this change to call by-name (which is probably required for the macro to work, but I'm not 100% sure) breaks binary compatibility so cannot be backported.

Reply:
yeah makes sense, i suspected i did it the other way round. fixed

Reply:
i think the call by name change was needed by something that i've since deleted. i have reverted it

Reply:
> yeah makes sense, i suspected i did it the other way round. fixed

The first line of the PR message still has the same phrasing.